zio-http/src/main/scala/zio/http/api/API.scala

```diff
@@ -115,6 +116,23 @@ final case class API[Input, Output](
   ): API.WithId[combiner.Out, Output, Id] =
     copy(input = self.input ++ in2).withId[Id]
 
+  /**
+   * Add a middleware spec into the API. Note that once you add the middleware,
+   * it returns a ServiceSpec (instead of API) disallowing it compose with any
+   * other APIs.
+   *
+   * If you have more than 1 API endpoints, compose them all together using
+   * `++`, and add `MiddlewareSpec` once and for all.
+   */
+  def middleware[MI, MO](middlewareSpec: MiddlewareSpec[MI, MO]): ServiceSpec[MI, MO, Id] =
+    ServiceSpec(self).middleware(middlewareSpec)
+
+  /**
+   * Convert API to a ServiceSpec.
+   */
+  def serviceSpec: ServiceSpec[Unit, Unit, Id] =
```

Comment:
Let's call this one `toServiceSpec` to match other ZIO naming conventions.

Reply:
Sure.